datasets/swda/swda.py

```diff
@@ -0,0 +1,199 @@
+turn/utterance-level dialog-act tags. The tags summarize syntactic, semantic, and pragmatic information about the
+associated turn. The SwDA project was undertaken at UC Boulder in the late 1990s.
+The SwDA is not inherently linked to the Penn Treebank 3 parses of Switchboard, and it is far from straightforward to
+align the two resources. In addition, the SwDA is not distributed with the Switchboard's tables of metadata about the
+conversations and their participants.
+"""
+
+# Homepage gathered from: https://github.com/cgpotts/swda#overview.
+_HOMEPAGE = "http://compprag.christopherpotts.net/swda.html"
+
+# More details about the license: https://creativecommons.org/licenses/by-nc-sa/3.0/.
+_LICENSE = "Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License"
+
+# Dataset main url.
+_URL = "https://github.com/NathanDuran/Switchboard-Corpus/raw/master/swda_data/"
+
+# Dialogue act integer labels. It encodes the actual act tags into integer values.
+act_label = {
+    "sd": 0,
+    "b": 1,
+    "sv": 2,
+    "%": 3,
+    "aa": 4,
+    "ba": 5,
+    "qy": 6,
+    "ny": 7,
+    "fc": 8,
+    "qw": 9,
+    "nn": 10,
+    "bk": 11,
+    "h": 12,
+    "qy^d": 13,
+    "bh": 14,
+    "^q": 15,
+    "bf": 16,
+    'fo_o_fw_"_by_bc': 17,
+    "na": 18,
+    "ad": 19,
+    "^2": 20,
+    "b^m": 21,
+    "qo": 22,
+    "qh": 23,
+    "^h": 24,
+    "ar": 25,
+    "ng": 26,
+    "br": 27,
+    "no": 28,
+    "fp": 29,
+    "qrr": 30,
+    "arp_nd": 31,
+    "t3": 32,
+    "oo_co_cc": 33,
+    "aap_am": 34,
+    "t1": 35,
+    "bd": 36,
+    "^g": 37,
+    "qw^d": 38,
+    "fa": 39,
+    "ft": 40,
+}
+
+
+class Swda(datasets.GeneratorBasedBuilder):
+    """
+    Switchboard Dialog Act Corpus
+    The Switchboard Dialog Act Corpus (SwDA) extends the Switchboard-1 Telephone Speech Corpus, Release 2,
+    with turn/utterance-level dialog-act tags. The tags summarize syntactic, semantic, and pragmatic information
+    about the associated turn. The SwDA project was undertaken at UC Boulder in the late 1990s.
+    """
+
+    # Splits url extensions for train, validation and test.
+    _URLS = {"train": _URL + "train_set.txt", "dev": _URL + "val_set.txt", "test": _URL + "test_set.txt"}
+
+    def _info(self):
+        """
+        Specify the datasets.DatasetInfo object which contains informations and typings for the dataset.
+        """
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types.
+            features=datasets.Features(
+                {
+                    "speaker": datasets.Value("string"),
+                    "utterance_text": datasets.Value("string"),
+                    "dialogue_act_tag": datasets.Value("string"),
+                    "dialogue_act_label": datasets.Value("int16"),
```

Comment:
You can define a ClassLabel feature type instead of having those two fields.
For example

```
"dialogue_act": datasets.ClassLabel(names=["sg", "b", "sv", etc.])
```

It means that the "sg" label correspond to 0, "b" corresponds to 1, "sv" correspond to 2 etc.

When yielding example in `_generate_examples` you can yield either the string label or the int label. Under the hood it will be stored as integers anyway.

Users will be able to know the mapping between the integers and the strings using `dataset.features["dialog_act"].names` which will return `["sg", "b", etc.]`.

There are also conversion methods `dataset.features["dialog_act"].str2int()` and `dataset.features["dialog_act"].int2str`().

Reply:
Had no idea about `datasets.ClassLabel` it made life so much easier! 😄